libass/ass_render.c

```diff
@@ -2730,12 +2750,17 @@ ass_render_event(ASS_Renderer *render_priv, ASS_Event *event,
     render_and_combine_glyphs(render_priv, device_x, device_y);
 
     memset(event_images, 0, sizeof(*event_images));
-    event_images->top = device_y - text_info->lines[0].asc;
-    event_images->height = text_info->height;
+    // VSFilter does *not* shift lines with a border > margin to be within the
+    // frame, so negative values for top and left may occur
+    event_images->top = device_y - text_info->lines[0].asc - text_info->border_top;
+    event_images->height =
+        text_info->height + text_info->border_bottom + text_info->border_top;
     event_images->left =
-        (device_x + bbox.x_min * render_priv->font_scale_x) + 0.5;
+        (device_x + bbox.x_min * render_priv->font_scale_x) + 0.5
+        - text_info->border_x;
```

Comment:
It's perfectly fine to write that in the same line, nobody works with 80-symbol terminals nowadays. I think that line widths up to ~120 are fine (you already have line with width nearly 120 in that PR anyway).

Reply:
Done.
*(The previous ~115 charater longer line was a comment with an URL Splitting this would imo not have been a good idea)*